packages/builder/src/builtin-tasks/run.ts

```diff
@@ -17,17 +22,29 @@ function filterNonExistent (scripts: string[]): string[] {
 }
 
 export async function runMultipleScripts (runtimeEnv: AlgobRuntimeEnv,
-  scriptNames: string[],
-  log: (...args: unknown[]) => unknown): Promise<void> {
+                                          scriptNames: string[],
+                                          log: (...args: unknown[]) => unknown,
+                                          persistSnapshots: boolean,
+                                          forceRun: boolean): Promise<void> {
   for (let i = 0; i < scriptNames.length; i++) {
     const scriptLocation = scriptNames[i];
-    log(
-      `Running script ${scriptLocation} in a subprocess so we can wait for it to complete`
-    );
+    const currentCP = loadCheckpoint(scriptLocation)
+    if (!forceRun && persistSnapshots && currentCP[runtimeEnv.network.name]) {
+      log(`Skipping: Checkpoint exists for script ${scriptLocation}`);
+      continue;
+    }
+    log(`Running script ${scriptLocation}`);
     await runScript(
       scriptLocation,
       runtimeEnv
     );
+    if (persistSnapshots) {
+      persistCheckpoint(
+        scriptLocation,
+        appendEnv(
+          currentCP,
+          runtimeEnv))
+    }
```

Comment:
We need to have a checkpoint class - and use it's object here. We will need to pass it to the deployed `script` as an argument - because user have to be able to add custom metadata.

Reply:
ah, now I see that we are in `algob run`. It would be still good to have same functionality in both `run` and `deploy`. In `run` we will not pass the checkopoint object to  the user script as an argument.

Reply:
Do you still want to have a class to handle this?

Reply:
I think it's better to have a class.

Reply:
Created a `Deployer` class, this class manages how checkpoints are mutated through multiple executions.